model/wire.cc

```diff
@@ -136,50 +133,41 @@ namespace minsky
     if (wp)
       dest.addWire(wp);
   }
-
-// For ticket 991. Adapted from https://www.stkent.com/2015/07/03/building-smooth-paths-using-bezier-curves.html
+ 
 namespace
 {
-	
-  vector<float> constructLowerDiagonalVector(int length) {
-    vector<float> result(length);
-
-    for (int i = 0; i < result.size() - 1; i++) {
-      result[i] = 1.0;
-    }
-    
-    result[result.size() - 1] = 2.0; 
-
-    return result;
-  }
-
-  vector<float> constructMainDiagonalVector(int n) {
-    vector<float> result(n);
-
-    result[0] = 2.0;
 
-    for (int i = 1; i < result.size() - 1; i++) {
-      result[i] = 4.0;
-    }
-    
-    result[result.size() - 1] = 7.0; 
-
-    return result;
+// For ticket 991. Construct tridoagonal matrix A which relates control points c and knots k (curved wire handles): Ac = k.
+ vector<vector<float>> constructTriDiag(int length) {
+
+   vector<vector<float>> result(length,vector<float>(length));
```

Comment:
This is not the most optimum way of doing this, but it will do for now.

Reply:
Would making it a void function and modifying the matrix in place be a better option?